arch/arm/core/cortex_m/Kconfig

```diff
@@ -35,6 +35,14 @@ config ISA_THUMB2
 	  technology is featured in the  processor, and in all ARMv7
 	  architecture-based processors.
 
+config CPU_CORTEX_M_HAS_VTOR
```

Comment:
HAS_VTOR is not right, STM32 does not has VTOR, it can remap the 0x0 physical address space with either the flash or the internal RAM.
A more coherent name would be CPU_CORTEX_M_CAN_RELOCATE for M0

Reply:
How about "HAS_FIXED_VT_ADDR" ?

Reply:
Yes for M0 only then

Reply:
I think CPU_CORTEX_M_HAS_VTOR is proper for native cortex vector relocation support - it also well References to the cortex manual. Looks like STMF0 has much different mechanism for vector relocation than VITOR is.

So maybe ther should be 2 configuration Keyword, on for informing that there is support relocation feature support, and 2nd for describe mechanism used (like VTOR is).

Reply:
I've checked most of the M0 SoCs spec datasheet (NXP L11XX/12XX etc.), it seems only STM32F0 support such a vector table remap mechanism.

All M0 SoC are all have fixed vector table address at 0x00000000, "HAS_FIXED_VT_ADDR" may not goot for this.

I think we use only one configuration is enough for this, enable it only in M0 SoCs but STM32F0.

Vector table relocation only enabled when SoC support bootlaoder, we can write like this:

In arch/arm/soc/{M0 SoCs}/Kconfig file:
select CPU_CORTEX_M0_VT_RELAY if (IS_BOOTLOADER || IMG_MANAGER)

As for STM32F0 Kconfig:
select CPU_CORTEX_M0_VT_REMAP if (IS_BOOTLOADER || IMG_MANAGER)
/* 2nd configuration, unnecessary, we can just leave it blank */

The issue https://github.com/zephyrproject-rtos/zephyr/issues/5651 can write linker like this:
linker script:
    ...
#ifdef CPU_CORTEX_M0
    . = 0x0;

#ifdef CPU_CORTEX_M0_VT_RELAY 
    KEEP(Relay VT)
    KEEP(Relay handler)
#else /* Some HW mechanism support vector table relocation like remap in STM32F0*/
    KEEP(Normal VT)
#endif 

    . = . + CONFIG_TEXT_SECTION_OFFSET
#else /* None M0 SoC, vector table can place any where */
    . = CONFIG_TEXT_SECTION_OFFSET;
    KEEP(Normal VT)
#endif 
    ....